patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+# Title
+
+Secure InnerSource
+
+# Patlet
+
+A collaborative InnerSource practice in a company benefits from a maximum level of shared source code, to allow participants from outside of the core team to contribute and re-use. In contrast to that, a functioning security practice demands to give the users the last amount of access to perform their job. By implementing various security measures and defining multiple sharing levels for different types of code, sharing and collaboration can be facilitated while minimizing the risks of code-leak and malicious code injection.
+
+# Problem
+
+A successful InnerSource program needs openness and transparency (e.g. access to code, issues, documentation, and roadmap), while good security practice is to minimize access, following the [principle of least privilege][principle-of-least-privilege]. How to balance and implement these seemingly contradicting requirements?
+
+# Story
+
+Most companies developing proprietary software will have source code that they do not want to leave the company, as this would harm their business. Think what would happen when the major competitors would have access to their latest features and would know what they are working on next. Even when the source code management system is not compromised from the outside, malicious employees or interns could leak code to the competition. Or think about gaining access to sensitive systems and services using credentials that are hard-coded in source code. Or injection of malicious code into a product’s codebase.
```

Comment:
```suggestion
Most companies developing proprietary software will have source code that they do not want to leave the company, as this may harm their business. Think what would happen when the major competitors would have access to their latest features and would know what they are working on next. Even when the source code management system is not compromised from the outside, malicious employees or interns could leak code to the competition. Or think about gaining access to sensitive systems and services using credentials that are hard-coded in source code. Or injection of malicious code into a product’s codebase.
```